src/js/game/gauges/throttle-meter.ts

```diff
@@ -1,53 +1,65 @@
 import Entity from "../../engine/entity";
 import Sprite from "../../engine/sprite";
-import Drawable from "../../engine/drawable";
+import { Drawable } from "../../engine/drawable";
 import EventSystem from "../../engine/eventSystem";
 import Position from "../../engine/position";
 import Engine from "../../engine/engine";
 import { Camera } from "../../engine/camera";
 import SpaceShip from "../ingameObjects/spaceship";
+import { PositionCordinates } from "../../engine/models/position.model";
+import { GetPositionCoordinates } from "../../engine/models/get-position-coordinates.model";
+import { GetDrawable } from "../../engine/models/get-drawable.mode";
 
-export class ThrottleMeter {
+export class ThrottleMeter implements GetDrawable, GetPositionCoordinates {
     engine: Engine = null;
     entity: Entity = null;
     drawable: Drawable = null;
     camera: Camera = null;
     position: Position = null;
     eventSystem: EventSystem = null;
     step = 1/4;
-    owner: SpaceShip; // is ship
+    owner: SpaceShip;
 
     constructor(engine: Engine) {
         this.engine = engine;
-        let sprite = new Sprite();
         this.entity = new Entity();
-        this.drawable = new Drawable();
+        this.drawable = this.getDrawable();
         this.position = new Position();
-        sprite.setImage(engine.loader.images.gaugeSheet.getImage());
-        this.drawable.bindSprite(sprite);
-        this.drawable.dimension = [44, 44];
-        this.drawable.size = [120, 120];
-        this.drawable.topLeft = [290,210];
         this.eventSystem = new EventSystem();
         this.camera = engine.camera;
     }
     
     bindOwner(owner: any) {
         this.owner = owner;
-        this.step = (4/this.owner.getMesh().power);
+        this.step = (4 / this.owner.getMesh().power);
     }
 
     update() {
-        console.log('throttle', this)
         let frameIndex = Math.floor(this.owner.throttle * this.step);
         this.drawable.topLeft = [290 + frameIndex * 48,210];
 
+        this.drawable.setPosition(this.getPositionCoordinates());
+    }
+
+    getPositionCoordinates(): PositionCordinates {
         const cp = this.camera.getPosition();
-        let newPosition = [
+        
+        let newPosition: PositionCordinates = [
```

Comment:
We can change PositionCoordinates with Vector type in future - since in fact it is a vector

Reply:
True!